expts/hydra-configs/datamodule/toymix.yaml

```diff
@@ -0,0 +1,76 @@
+module_type: "MultitaskFromSmilesDataModule"
+args:
+  prepare_dict_or_graph: pyg:graph
+  featurization_n_jobs: 30
+  featurization_progress: True
+  featurization_backend: "loky"
+  processed_graph_data_path: "../datacache/neurips2023-small/"
+  num_workers: 30 # -1 to use all
+  persistent_workers: False
+  task_specific_args:
+    qm9:
+      df: null
+      df_path: ${constants.data_dir}/qm9.csv.gz
+      # wget https://storage.googleapis.com/graphium-public/datasets/neurips_2023/Small-dataset/qm9.csv.gz
+      # or set path as the URL directly
+      smiles_col: "smiles"
+      label_cols: ["A", "B", "C", "mu", "alpha", "homo", "lumo", "gap", "r2", "zpve", "u0", "u298", "h298", "g298", "cv", "u0_atom", "u298_atom", "h298_atom", "g298_atom"]
+      # sample_size: 2000 # use sample_size for test
+      splits_path: ${constants.data_dir}/qm9_random_splits.pt  # Download with `wget https://storage.googleapis.com/graphium-public/datasets/neurips_2023/Small-dataset/qm9_random_splits.pt`
+      seed: ${constants.seed} #*seed
+      task_level: graph
+      label_normalization:
+        normalize_val_test: True
+        method: "normal"
+
+    tox21:
+      df: null
+      df_path: ${constants.data_dir}/Tox21-7k-12-labels.csv.gz
+      # wget https://storage.googleapis.com/graphium-public/datasets/neurips_2023/Small-dataset/Tox21-7k-12-labels.csv.gz
+      # or set path as the URL directly
+      smiles_col: "smiles"
+      label_cols: ["NR-AR", "NR-AR-LBD", "NR-AhR", "NR-Aromatase", "NR-ER", "NR-ER-LBD", "NR-PPAR-gamma", "SR-ARE", "SR-ATAD5", "SR-HSE", "SR-MMP", "SR-p53"]
+      # sample_size: 2000 # use sample_size for test
+      splits_path: ${constants.data_dir}/Tox21_random_splits.pt  # Download with `wget https://storage.googleapis.com/graphium-public/datasets/neurips_2023/Small-dataset/Tox21_random_splits.pt`
+      seed: ${constants.seed}
+      task_level: graph
+
+    zinc:
+      df: null
+      df_path: ${constants.data_dir}/ZINC12k.csv.gz
+      # wget https://storage.googleapis.com/graphium-public/datasets/neurips_2023/Small-dataset/ZINC12k.csv.gz
+      # or set path as the URL directly
+      smiles_col: "smiles"
+      label_cols: ["SA", "logp", "score"]
+      # sample_size: 2000 # use sample_size for test
+      splits_path: ${constants.data_dir}/ZINC12k_random_splits.pt  # Download with `wget https://storage.googleapis.com/graphium-public/datasets/neurips_2023/Small-dataset/ZINC12k_random_splits.pt`
+      seed: ${constants.seed}
+      task_level: graph
+      label_normalization:
+        normalize_val_test: True
+        method: "normal"
+  featurization:
```

Comment:
Do you mean to move featurization somewhere else? Not sure I understand.

Reply:
I meant about the `featurization: ` being part of `toymix.yaml` rather than the model file. Because the featurization will impact the model.

Reply:
Okay, so I had a closer look and here are my thoughts: It is true that `featurization` impacts the model. However, the settings there appear to be model-agnostic nonetheless. If stored with the model, and multiple models use the same featurization, this would also mean that the featurization configs are duplicated across those model definitions. I also checked the neurips configs and there we use the same featurization for different models.

Note that with `hydra`, users can compose their own configs based on the granularity that best suits their application. E.g., if I were to experiment with different PE/SE then it would make sense to further extract featurization and PE/SE definitions such that I can easily switch them via the CLI.